rsciio/sur/specifications.yaml

```diff
@@ -1,4 +1,5 @@
-format_name: Digital Surf Surface
+format_name: DigitalSurfSurface
+format_name_aliases: Digital Surf Surface
```

Comment:
In https://github.com/hyperspy/hyperspy/pull/3009 I added support for either string or list of strings.

Reply:
I would allow only list as this would make it more consistent and more simple to use because string are iterable and can be indexed which can be confusing if both list and string are allowed. What do you think?

Reply:
How would we enforce that it is a list, I guess just in the review process?
Checking if something is a string instead of a list is not so hard in any case.

Reply:
We need to test all possible values of the keyword argument `reader` and remove string support by removing  https://github.com/jlaehne/hyperspy/blob/b78e1417ed91dc7cf420f0720095974347f2f5d8/hyperspy/io.py#L60-L62

Reply:
> Checking if something is a string instead of a list is not so hard in any case.

Indeed, and this is more to simplify the API and avoid supporting different type for the key `format_name_aliases`.

Reply:
OK, I added tests for all entries of `specifications.yaml` to check if they are of the correct type. `format_name_aliases` being the only optional one.

Reply:
Is it worth checking that the `format_name` and aliases themselves?
For example

```python
@pytest.mark.parametrize("reader", [None, "DigitalSurfSurface", "Digital Surf Surface"])
def test_format_name_aliases(reader):
   s = hs.load(fname, reader=reader)
```

Reply:
It is done for two types of files (`hspy` and `emd`) in [test_io.py](https://github.com/hyperspy/rosettasciio/pull/35/files#diff-64b71f8383d9b95c346faf7d851afeb1522215deca9900e5467683c1b700e068)
One with a single, and one with multiple aliases. Plus for different capitalization.

I have chosen file formats that can also be written. Otherwise we need to rely on saved test files, but I think it is not necessary to test every single alias that would be possible if we test that the mechanism works in principle and that the files are defined consistently.